src/benchmarklib/ssb/ssb_table_generator.cpp

```diff
@@ -0,0 +1,62 @@
+#include "ssb_table_generator.hpp"
+
+#include "external_dbgen_utils.hpp"
+#include "storage/table.hpp"
+
+namespace hyrise {
+
+const auto ssb_table_names = std::vector<std::string>{"part", "customer", "supplier", "date", "lineorder"};
+
+SSBTableGenerator::SSBTableGenerator(const std::string& dbgen_path, const std::string& data_path, float scale_factor,
+                                     ChunkOffset chunk_size)
+    : SSBTableGenerator(dbgen_path, data_path, scale_factor, create_benchmark_config_with_chunk_size(chunk_size)) {}
+
+SSBTableGenerator::SSBTableGenerator(const std::string& dbgen_path, const std::string& data_path, float scale_factor,
+                                     const std::shared_ptr<BenchmarkConfig>& benchmark_config)
+    : AbstractTableGenerator(benchmark_config),
+      FileBasedTableGenerator(benchmark_config, data_path + "/"),
+      _dbgen_path(dbgen_path),
+      _scale_factor(scale_factor) {}
+
+std::unordered_map<std::string, BenchmarkTableInfo> SSBTableGenerator::generate() {
+  generate_csv_tables_with_external_dbgen(_dbgen_path, ssb_table_names, "resources/benchmark/ssb/schema", _path,
+                                          _scale_factor, "-T a");
+
+  // Having generated the .csv files, call the FileBasedTableGenerator just as if those files were user-provided.
+  const auto& generated_tables = FileBasedTableGenerator::generate();
+
+  // FileBasedTableGenerator automatically stores a binary file. Remove the CSV data to save some space.
+  remove_csv_tables(_path);
```

Comment:
If there is not a third call that's difference later on: execute the CSV removal in the `generate()` function?

Reply:
you mean in `FileBasedTableGenerator::generate()`? I would not do so. If `FileBasedTableGenerator` removed all CSVs, it removes its source data. In the worst case, this would mean we would have to, e.g., download IMDB data multiple times if we change sth with the binary parser, or we could only remove the files if binary data is cached, ...